src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public void setOperation(String operation) {
+        Optional<Operation> matchingOperation = Arrays.stream(Operation.values())
+                .filter(val -> val.getCode().equals(operation))
+                .findFirst();
+
+        if (matchingOperation.isPresent()) {
+            this.operation = matchingOperation.get();
+        } else {
+            throw new IllegalArgumentException("Invalid operation code: " + operation);
+        }
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+    }
```

Comment:
The `Operation` enum is well implemented with a private field and a constructor. However, it's a good practice to place enums in separate files unless they are very tightly coupled with the class they are declared in. This ensures better separation of concerns and makes the codebase more maintainable.